azurelinuxagent/common/osutil/default.py

```diff
@@ -1046,20 +1046,24 @@ def device_for_ide_port(self, port_id):
         device = None
         path = "/sys/bus/vmbus/devices/"
         if os.path.exists(path):
-            for vmbus in os.listdir(path):
-                deviceid = fileutil.read_file(os.path.join(path, vmbus, "device_id"))
-                guid = deviceid.lstrip('{').split('-')
-                if guid[0] == g0 and guid[1] == "000" + ustr(port_id):
-                    for root, dirs, files in os.walk(path + vmbus):
-                        if root.endswith("/block"):
-                            device = dirs[0]
-                            break
-                        else : #older distros
-                            for d in dirs:
-                                if ':' in d and "block" == d.split(':')[0]:
-                                    device = d.split(':')[1]
-                                    break
-                    break
+            try:
+                for vmbus in os.listdir(path):
+                    deviceid = fileutil.read_file(os.path.join(path, vmbus, "device_id"))
+                    guid = deviceid.lstrip('{').split('-')
+                    if guid[0] == g0 and guid[1] == "000" + ustr(port_id):
+                        for root, dirs, files in os.walk(path + vmbus):
```

Comment:
If the deviceid is malformed, it's possible that guid may be a list with only one member, in which case referencing guid[1] will raise IndexError.